consensus/istanbul/backend/val_enode_share.go

```diff
@@ -0,0 +1,249 @@
+// Copyright 2017 The Celo Authors
+// This file is part of the celo library.
+//
+// The celo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The celo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the celo library. If not, see <http://www.gnu.org/licenses/>.
+
+package backend
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/istanbul"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// ==============================================
+//
+// define the validator enode share message
+
+type sharedValidatorEnode struct {
+	Address  common.Address
+	EnodeURL string
+	View     *istanbul.View
+}
+
+type valEnodeShareMessage struct {
```

Comment:
Could you use the istanbul.Message type to store your new message.  Could you also please refactor the `Announce` message to also use it. 🙏

That type used to be a private type within istanbul.core package, but Asa recently made it a public type in the istanbul package, specifically so that Announce and related messages can use it.